### PR TITLE
Fixing checklist for Buffer Overflow, Integer Error, Input Validation, and Encapsulation modules

### PR DESCRIPTION
**Pull request by shanioffe**
For all security injections containing checklists, progress bars and highlights on correct answers were added.

If you're avoiding this review because there's 107 files changed, it's actually just 24 security injections. If you clone this repo, pull my branch, and run `node create.js` and then `open .` you'll get a folder of HTML links to the security injections that you can open up and try in your browser. The folder will show all of our security injections, but only the 24 with checklists will rebuild (only will be Buffer Overflow, Integer Error, Input Validation, or Encapsulation).

Look out for:
- The progress bar filling logically: when correct answer is clicked, progress bar should fill, and not advance to next question until full. Note: progress bar is removed only for questions that ask user to type in a range.
- Correct answers should highlight green: highlighted answers should reset after each question so you aren't seeing correct answers of a previous section, they should be large enough to click without difficulty.
- **The correct answers match up with the question being asked:** please pay extra attention to this.

### CODE REVIEW

File: content/Secure Coding/Buffer Overflow/CS2 Java/checklist.js [hunk 1]
```diff
@@ -1,134 +1,163 @@
...
+						resetPopOver();
+						times = 1;
+						showPopOver(spans[spanLoc2]);
+					} else {
+						resetPopOver();
+						$(".span-" + name).click(); //click the empty span so that next question is asked.
+					}
+				} else {
+					//$('.popover').css('background-color', 'lightpink');
 				}
```
Comment (JosueMolinaMorales) [inline content/Secure Coding/Buffer Overflow/CS2 Java/checklist.js]: Delete commented out code if not used.

**Review by JosueMolinaMorales — APPROVED** [review]
Looks good! Just a small comment that was left behind.

**Review by pzalep1 — APPROVED** [review]
Looks good!

**Review by jouypo23 — APPROVED** [review]
Nice work!
goodjob

**Review by Cwagne17 — APPROVED** [review]

**Review by pmanal1 — APPROVED** [review]

**Pull request merged**